test/powershell/Language/Interop/DotNet/DotNetInterop.Tests.ps1

```diff
@@ -180,7 +225,95 @@ namespace DotNetInterop
         }
     }
 
-    It "Set access of an indexer that accepts ByRef-like type should fail gracefully" {
-        { $testObj[1] = 1 } | Should -Throw -ErrorId "InvalidCastToByRefLikeType"
+    It "Set access of an indexer that accepts ByRef-like type value should fail gracefully" {
+        { $testObj[1] = 1 } | Should -Throw -ErrorId "CannotIndexWithByRefLikeReturnType"
+    }
+
+    Context "Passing value that is implicitly/explicitly castable to ByRef-like parameter in method invocation" {
+        
+        BeforeAll {
+            $typeXml = @'
+<Types>
+    <Type>
+        <Name>DotNetInterop.Test2</Name>
+        <Members>
+            <CodeMethod>
+                <Name>RunTest</Name>
+                <CodeReference>
+                    <TypeName>DotNetInterop.CodeMethods</TypeName>
+                    <MethodName>RunMethod</MethodName>
+                </CodeReference>
+            </CodeMethod>
+            <CodeProperty>
+                <Name>TestName</Name>
+                <GetCodeReference>
+                    <TypeName>DotNetInterop.CodeMethods</TypeName>
+                    <MethodName>GetProperty</MethodName>
+                </GetCodeReference>
+                <SetCodeReference>
+                    <TypeName>DotNetInterop.CodeMethods</TypeName>
+                    <MethodName>SetProperty</MethodName>
+                </SetCodeReference>
+            </CodeProperty>
+        </Members>
+    </Type>
+</Types>
+'@
+            $typeXmlFile = Join-Path -Path $TestDrive -ChildPath "TestType.ps1xml"
+            Set-Content -Path $typeXmlFile -Value $typeXml -Encoding Ascii
+            $ps = [powershell]::Create()
+            $ps.AddCommand("Update-TypeData").AddParameter("AppendPath", $typeXmlFile).Invoke()
```

Comment:
Update-TypeData can be used without creating a file if you'd prefer.

Reply:
Indeed - much easier. Or is there some reason to use the file?

Reply:
No particular reason, I just went with the file without considering other options ;) I will change to simply the test.

Reply:
Updated to use `Update-TypeData -TypeName -MemberType ...`